pkg/geo/geomfn/distance.go

```diff
@@ -775,3 +775,22 @@ func HausdorffDistanceDensify(a, b geo.Geometry, densifyFrac float64) (*float64,
 	}
 	return &distance, nil
 }
+
+// NearestPoints returns the first point located on geomentry A on the shortest line between geometries.
+func NearestPoints(a, b geo.Geometry) (geo.Geometry, error) {
+	if a.SRID() != b.SRID() {
+		return geo.Geometry{}, geo.NewMismatchingSRIDsError(a.SpatialObject(), b.SpatialObject())
+	}
+	if a.Empty() || b.Empty() {
+		return geo.Geometry{}, nil
+	}
+	nearestPoint, err := geos.NearestPoints(a.EWKB(), b.EWKB())
```

Comment:
sorry to do this, but going through this I genuinely think it's nicer to reuse or distance implementation.

GEOS has different edge case handling to us, particularly towards POINT EMPTY and I'd rather we handle this in a consistent way

It's should be as simple as re-using  https://github.com/cockroachdb/cockroach/blob/master/pkg/geo/geomfn/distance.go#L126 (or adapting `distanceLineStringInternal` to return a `*geom.LineString` and downstream upstream functions to use `NewGeometryFromGeomT`).